src/main/java/dev/jbang/ExtendedRunUnit.java

```diff
@@ -0,0 +1,264 @@
+package dev.jbang;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.*;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * This class wraps a reference to an runnable/executable resource in the form
+ * of a RunUnit. It also holds all additional information necessary to be able
+ * to actually run/execute that resource. This is information that can not be
+ * induced or extracted from the resource itself but is information that is
+ * provided by the user or by the environment at runtime.
+ *
+ * This class also implements RunUnit, passing all calls directly to the wrapped
+ * RunUnit object. This makes it easier to use this class in places where it's
+ * not really important to know with what type or RunUnit we're dealing.
+ */
+public class ExtendedRunUnit implements RunUnit {
```

Comment:
Just one example but there are dozens:

Info.java:48

```
			javaVersion = xrunit.javaVersion();
```

Meaning it will return the Java version requested by the Scxript or Jar without having to know if what you have is a Script or a Jar.

Otherwise you either have to do (stupidly):

```
if (xunit.runUnit instanceof Script) {
   javaVersion = xrunit.script().javaVersion();
} else {
   javaVersion = xrunit.jar().javaVersion():
}
```

or add a new helper method just for this, eg:

```
javaVersion = xrunit.unit().javaVersion();
```

Personally I prefer my solution over this last one.

A completely different solution, which work too, is to just split `ExtendedRunUnit` off from the wrapped object inside it. So instead of passing a single `Extended RunUnit` which wraps a `Script` or `Jar` object we could pass two object everytwhere: one `RunUnit` (script or Jar) and one ... `ExtraInfoSuppliedByUser` object. But I'm not positive this would be any better.